### PR TITLE
ZCS-2985, ZCS-3097, ZCS-3101, ZCS-3102, ZCS-3250: Fixed visibility of account/resource email and account/resource/domain description, Removed "required" property from zimbra(Admin)AuthTokenLifetime of account, Fixed selecting ZIMBRA as F/B Interop Provider & Fixed moving alias

### PR DESCRIPTION
**Pull request by k-kato**
Fixed account and alias related issues

### CODE REVIEW

File: WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js [hunk 1]
```diff
@@ -138,10 +138,14 @@ ZaFp.getProviders = function () {
 //@prefix: foreign principal prefixes
 ZaFp.findDupPrefixFp = function (entries, prefix) {
     for (var i =0 ; i < entries.length; i ++) {
-        if (entries[i].indexOf(prefix) == 0) {
+        // if a new provide is ZIMBRA, the prefix is empty string ""
+        if(prefix.length == 0 && entries[i].indexOf(":") == -1) {
+            return true;
+        }
+        if(prefix.length != 0 && entries[i].indexOf(prefix) == 0) {
```
Comment (ykomiyam) [inline WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js]: Is it intentionally declared two "if" clauses separately?  If not why don't you put two conditions into one "if"?
And put a space between "if" and "(" to follow the Zimbra's coding conventions.

Comment (k-kato) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js]: Fixed it.

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js [hunk 2]
```diff
@@ -1499,7 +1499,7 @@ ZaAccountXFormView.myXFormModifier = function(xFormObject, entry) {
         headerItems.push({type:_OUTPUT_, ref:ZaAccount.A_zimbraMailTransport, labelLocation:_LEFT_,label:ZabMsg.attrDesc_mailHost});
     }
 
-    if (ZaItem.hasReadPermission(ZaAccount.A_name, entry)) {
+    if (entry[ZaAccount.A_name]) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js]: AjxUtil.isEmpty would be better check

File: WebRoot/js/zimbraAdmin/resource/view/ZaResourceXFormView.js [hunk 3]
```diff
@@ -530,8 +530,9 @@ ZaResourceXFormView.myXFormModifier = function(xFormObject, entry) {
 	    headerItems.push({type:_OUTPUT_, ref:ZaResource.A_mailHost, labelLocation:_LEFT_,label:ZaMsg.NAD_MailServer, getDisplayValue: AjxUtil.htmlEncode});
     }
 	
-    if (ZaItem.hasReadPermission(ZaResource.A_name, entry))
+    if (entry[ZaAccount.A_name]) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/resource/view/ZaResourceXFormView.js]: same as above

File: WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js [hunk 4]
```diff
@@ -138,10 +138,14 @@ ZaFp.getProviders = function () {
 //@prefix: foreign principal prefixes
 ZaFp.findDupPrefixFp = function (entries, prefix) {
     for (var i =0 ; i < entries.length; i ++) {
-        if (entries[i].indexOf(prefix) == 0) {
+        // if a new provide is ZIMBRA, the prefix is empty string ""
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js]: typo: provider

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js (hunk 2 above)
Comment (k-kato) [reply WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js]: @silentsakky I found AjxUtil.isEmailAddress. It seems to be better to check if the value of entry[ZaAccount.A_name] is an email address (string).
Which do you think is better?
- if (!AjxUtil.isEmpty(entry[ZaAccount.A_name])) {
- if (AjxUtil.isEmailAddress(entry[ZaAccount.A_name], false)) {
    // set false to check strictly

Comment (silentsakky) [reply WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js]: looks good, we can use it

Comment (k-kato) [reply WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js]: fixed to use AjxUtil.isEmailAddress

File: WebRoot/js/zimbraAdmin/resource/view/ZaResourceXFormView.js (hunk 3 above)
Comment (k-kato) [reply WebRoot/js/zimbraAdmin/resource/view/ZaResourceXFormView.js]: fixed to use AjxUtil.isEmailAddress

File: WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js (hunk 4 above)
Comment (k-kato) [reply WebRoot/js/zimbraAdmin/accounts/model/ZaFp.js]: fixed the typo

**Pull request merged**